Bio/PDB/rotamers.py

```diff
@@ -0,0 +1,279 @@
+"""
+A tool that introduces a point mutation to a given residue based on the Dunbrack rotamer library.
+The tool uses sample residues stored in small PDB files, and uses a reduced rotamer library
+to save space (only rotamers with higher than 5% probability are present).
+The tool makes in place modification on a PDB structure object.
+"""
+
+from Bio.Entrez.Parser import ValidationError
+from Bio.PDB import Polypeptide
+from Bio.PDB.NeighborSearch import NeighborSearch
+from Bio.PDB.Selection import unfold_entities
+from Bio.PDB.vectors import calc_dihedral, Vector, rotaxis2m
+from Bio import SVDSuperimposer
+from collections import defaultdict
+from Bio.PDB.ic_data import CHI_ANGLES, RESIDUE_ORDER, VW_RADII
+import numpy as np
+import os
+import gzip
+
+
+DATA_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "rotamers")
+DUNBRACK_FILTERED = f"{DATA_DIR}/dunbrack_reduced.gz"
+
+
+class Rotamer:
+    """
+    Object to store rotamers
+    """
+
+    def __init__(self, resname, psi, phi, chi1, chi2, chi3, chi4, probability):
+        """
+        Init the rotamer with given diherdral angles
+        """
+        self.psi = float(psi)
+        self.phi = float(phi)
+        self.chi1 = float(chi1)
+        self.chi2 = float(chi2)
+        self.chi3 = float(chi3)
+        self.chi4 = float(chi4)
+        self.resname = resname
+        self.probability = float(probability)
+        self.sample_residue = False
+
+    def __getitem__(self, item):
+        """
+        Returns the given angle property of the rotamer
+        """
+        return getattr(self, item.lower())
+
+    def set_sample_residue(self, sample_residue):
+        """
+        Sets a sample residue for the rotamer object
+        This function is only called during RotamerSampling to save memory
+        """
+        self.sample_residue = sample_residue
+
+    def get_sample_residue(self):
+        """
+        Returns the stored sample residue
+        """
+        if self.sample_residue:
+            return self.sample_residue
+        raise ValueError("Sample residue is not set, use RotamerSampling to set it!")
+
+
+class RotamerSampling:
```

Comment:
`RotamerSampling` feels a bit strange as a class name in English. I think usually you want a name that is like an object or something that does an action, not the action itself. Something like `RotamerSampler` might be better, but I'm not familiar enough with these specific biological transformations to say if that makes sense. Maybe someone with more bio knowledge can comment?

Reply:
The name was suggested by @JoaoRodrigues. I am happy to change if needed!